library/src/lib/calendar/calendar-views/calendar-year-view/calendar-year-view.component.ts

```diff
@@ -0,0 +1,155 @@
+import { Component, OnInit, ViewEncapsulation, Output, Input, EventEmitter, ElementRef, AfterViewChecked } from '@angular/core';
+import { FdDate } from '../../models/fd-date';
+
+/** Component representing the YearView of the Calendar Component. */
+@Component({
+    selector: 'fd-calendar-year-view',
+    templateUrl: './calendar-year-view.component.html',
+    styleUrls: ['./calendar-year-view.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    host: {
+        '[attr.id]': 'id + "-year-view"'
+    }
+})
+export class CalendarYearViewComponent implements AfterViewChecked, OnInit {
+
```

Comment:
this file needs to be better organized. Inputs with inputs, outputs with outputs etc